### DISCUSSION TITLE
Lower-Touch Defender for Endpoint Onboarding for Android Devices - odds+endpoints

### DISCUSSION THREAD

**Discussion by giscus** [ennnbeee/oddsandendpoints-web#11] [Comments]
# Lower-Touch Defender for Endpoint Onboarding for Android Devices - odds+endpoints

Using Intune and OEMConfig for Samsung Knox Android Enterprise devices to improve the onboarding of devices to Defender for Endpoint.

https://www.oddsandendpoints.co.uk/posts/android-enterprise-defender-onboarding/

**Comment by Niko2238** [comment]
Great article! Thanks for the write up. To clarify there is currently no equivalent to the Knox Service Plugin for Google Pixel phones?

**Reply by ennnbeee** [reply]
My current understanding is that Google do not provide an OEMConfig app for their Pixel phones; Zebra, Datalogic, Lenovo, Honeywell, Nokia, and Samsung do, but not Google directly.

**Comment by ReaperSTG** [comment]
Hello,

I have tried the steps but the Samsung Knox Service Plugin App reports the following error:
[23000] ‘Battery optimisation alllowlist’ couldn't be set to ‘com.microsoft.scmx’ in device policies. (Application: com.microsoft.scmx, Error: ERROR_INVALID_INPUT).

We use Samsung devices rolled out in Intune (COPE) in conjunction with Samsung KME. We have tried the way officially suggested by Microsoft to integrate Low Touch but it did not simplify things at all, so we wanted to try it via KSP as noted in the article, but there seem to be problems here.

Is there anything known?

**Reply by ennnbeee** [reply]
I would double check where you are applying the battery optimisation policy, the setting in this article that worked on my test devices is under 'Device-Wide' as per the video.

**Reply by ReaperSTG** [reply]
Hey!
i checked the settings several times. 
1) As it seems deploying Low Touch Device Configuration only works for Fully managed or WP-C enrolled devices, 
not for devices in COPE mode (info seen in the configuration settings in Intune)! 

_**Device-wide policies (Selectively applicable to Fully Manage Device (DO) or Work Profile-on company owned devices (WP-C) mode as noted)**_

2) The error mentioned in my previous post actually seems to be an error while the other entries in the KSP app are merely indications that the settings have no effect due to the WP-C mode which is nonsens because COPE is not the same as WP-C mode.
So for my understanding, either information in Intune is wrong or the one the KSP App offers. 

_****_**_[90022]: Permission controls...set in device-wide policy. However this policy is not supported in WP-C mode (device side), and it therfore will have no effect._**_****_

**Reply by ennnbeee** [reply]
Ahh I  get what you're saying now, looks like the tooltip for battery optimisation settings under the Device-Wide settings points you in the direction of the Work-Profile specific area of the OEMConfig:

*"Use this control to exempt applications from battery usage optimizations such as Android Doze mode. For a fully managed device with a Work profile, enter the list of application on the personal profile to allowlist. To specify Work profile-only apps, go to Work Profile Policies > App Management section. Enter a comma-separated list of package names to specify the apps to allowlist."*

Sadly though the Work-Profile Battery optimization allowlist is under a premium Application Management policy setting, so you'll need a Knox license key.

They were making these available for free, but not sure if that's still the case.

**Reply by ReaperSTG** [reply]
Hey!
Thanks for your suggestion!
I think we´re mixing things up - sry for that!

i´m facing two issues....

1) The original made settings - like you suggested - throw the error with the "invalid syntax" for battery optimization.
If this gets fixed by setting it up within the Work Profiles Policies, i´m happy with that but it would confuse me, to be honest.

2) About the WP-C issue this is related to the permission control (appear on top/All files access) settings and maybe i´m blind but i can´t find anything like that in the Work Profile Policies section.

**Reply by Stanzoheetik** [reply]
Hi Raeper, 

Im am currently exactly experiencing what you describe. Almost all our Android device in our org are enrolled using intune with enrollment 'Android Enterprise corporate-owned (COPE)'. We get the same error [23000] ‘Battery optimisation alllowlist’ couldn't be set to ‘com.microsoft.scmx’ in device policies. (Application: com.microsoft.scmx, Error: ERROR_INVALID_INPUT).

Ive set the policy in Work profile policies (Profile Owner) -> Application management policies (Premium) and that seems to work for the battery optimisation policy. I have not found the options for the other policies

**Reply by ennnbeee** [reply]
@ReaperSTG regarding point 2, the Permissions Controls don't seem to be an option within the WP-C section of the KSP settings, so luckily you're not blind, they just don't exist as part of the app management controls.

**Reply by TheITRx** [reply]
| they just don't exist as part of the app management controls.

this is so ironic. :( "Company-Owned Company profile". Yet we are not able to auto-grant permissions required by our security app. I think it would make sense for personal devices. but not WP-C.

**Comment by deweez** [comment]
I managed to replicate most of the above with the Motorola OEMConfig toolset, but weirdly they make you provide the SHA256 for the app to grant All Files Access to Defender, and the one on apkmirror didn't seem to work. Either way, here's the settings I used in Intune:

Debug tools policies > Debug Mode > Allow debug mode: True
System policies > Battery non-optimized apps > Allow non-battery optimized applications: True
System policies > Battery non-optimized apps > Non-battery optimized application list: com.microsoft.scmx
System policies > Display over the other apps settings > Enable display over the other apps: True
System policies > Display over the other apps settings > Display Over The Other Apps list > click the Three Dots next to the Display Over The ... entry in the tree, then click Add setting, then click the new value in the tree > App package name: com.microsoft.scmx
System policies > All Files Access > Enable All Files policy: True
System policies > All Files Access > All Files apps list > click the Three Dots next to the All Files apps list entry in the tree, then Add setting, then click the new value in the tree > App Package name: com.microsoft.scmx / App certificate: Unable to find valid cert

**Reply by ennnbeee** [reply]
Thanks for this, I've updated the post to include these settings, what a legend your are mate.

**Comment by Nico3869** [comment]
Hi. Great article ! It works fine on Samsung devices for most of permissions requests but have you found a way to auto-approve permissions for the accessibility service ?

**Reply by ennnbeee** [reply]
@Nico3869 even in the [Samsung docs](https://docs.samsungknox.com/admin/knox-platform-for-enterprise/knox-service-plugin/configure-advanced-policies/deep-settings-customization/#configurable-deep-settings) there doesn't appear to be a reference to the accessibility service settings, so I don't think we'll be able to sort out this final permission using Intune.

**Reply by Jordinthecloud** [reply]
Yea, unfortunately we're stuck on the same thing. Great article though! Works like a charm.

**Reply by Jordinthecloud** [reply]
The Access device location permissions are not approved on my side, any thoughts on this?

**Reply by ennnbeee** [reply]
@Jordinthecloud assuming these were set in the App Config profile, I wonder if something has changed with location permissions. I'll see if I can test this week.

**Reply by ennnbeee** [reply]
@Jordinthecloud re-tested, everything seemed OK, doubled down on the permissions for location and set the following:

`Location access (fine)`
`Location access (coarse)`
`Location access (background)`

Got a notification that location permissions were granted, and ticks for location in the Defender setup.

**Reply by Jordinthecloud** [reply]
I already have those set. Still same issue on my side. 

I have to add that I disabled location sharing during enrollment. But that wouldn’t stop Defender from granting permissions.

**Comment by TheITRx** [comment]
Very nice article. I wish the official once were as detailed as this. THank yoU! Also worth mentioning that it appears WP-C devices doesn't support Permission Controls > Permission Configuration > All files access
Appear on top

**Comment by naybahood** [comment]
Thanks for the guide!
How can i make knox service plugin to apply latest policies automatically?
I have to open the app and select apply latest policies. Or do i have to wait for some time to let the KSP app apply it automatically?

**Reply by ennnbeee** [reply]
Intune _should_ push config changes to devices as part of the normal check-in cycle, you can always force this issue by changing the name/version of the OEMConfig policy within the KSP settings.
